src/index.ts

```diff
@@ -25,14 +36,32 @@ export default class SceneWriter {
   emitCode(): string {
     let code = ''
     this.entities.forEach((entity, name) => {
-      code += this.writeEntity(name) + '\n'
+      code += this.writeEntity(entity, name) + '\n'
     })
     return code
   }
 
-  private writeEntity(name: string): string {
+  private writeEntity(entity: DCL.Entity, name: string): string {
     let code = ''
     code += `const ${name} = new Entity()\n`
+
+    const parent = entity.getParent()
+    if (parent) {
+      let parentEntity: KeyValuePair<string, DCL.Entity>
+
+      this.entities.forEach((ent, name) => {
+        if (ent === parent) {
+          parentEntity = { key: name, value: ent }
+        }
+      })
```

Comment:
Please move this to a function with a good name, it is not clear why you store a KeyValuePair instead of a plain string